Mathlib/CategoryTheory/Limits/ConcreteCategory/WithAlgebraicStructures.lean

```diff
@@ -77,4 +106,356 @@ lemma colimit_no_zero_smul_divisor
 
 end module
 
+section module_over_filtered_colimit
+
+/-!
+# Module structure of `colim Mᵢ` over `colim Rᵢ`
+
+Let `R` be the filtered colimit of rings `{Rⱼ}` and `M` be the filtered colimit of abelian groups
+`{Mⱼ}`  with `j` running through the same filtered index category `J`. If for each `j ∈ J`, `Mⱼ` is
+an `Rⱼ`-module such that the `Rⱼ`-actions are compatible, then `M` is an `Rⱼ`-module for all `j`
+and `M` is an `R`-module. We put a module structure on `colim Mᵢ` over `colim Rᵢ` where the scalar
+multiplication is given by `r • m = [a|ₛ • b|ₛ]` where `a : Rᵢ` and `b : Mⱼ` represent `r` and `m`
+and `s` refines both `i` and `j`.
+
+-/
+
+variable {ℜ𝔦𝔫𝔤 : Type u} [Category.{u'} ℜ𝔦𝔫𝔤] [ConcreteCategory.{max uc₁ w} ℜ𝔦𝔫𝔤]
```

Comment:
Can you say anything about what the heck is going on with these universes? My naive expectation would be that if (R : Type u) is a category then it would forget to Type u.

Reply:
This is copied from concrete category file, I am not too sure the motivation, but it is certainly more general than forgeting to type `u`